assets/components/payPalExpressButton/payPalExpressButton.jsx

```diff
@@ -5,40 +5,58 @@
 import React from 'react';
 import { connect } from 'react-redux';
 
-import {
-  setupPayPalExpressCheckout,
-} from 'helpers/payPalExpressCheckout/payPalExpressCheckoutActions';
+import * as payPalExpressCheckout from 'helpers/payPalExpressCheckout/payPalExpressCheckout';
 
+import type { IsoCurrency } from 'helpers/internationalisation/currency';
 
 // ---- Types ----- //
 
 type PropTypes = {
-  setupPayPalCheckout: Function,
+  dispatch: Function,
   callback: Function,
+  csrfToken: string,
+  amount: number,
+  billingPeriod: string,
+  currency: IsoCurrency,
 };
 
 
-// ----- Component ----- //
+function setupPayPalExpressCheckout(
+  csrfToken: string,
+  amount: number,
+  billingPeriod: string,
+  currency: IsoCurrency,
+  callback: Function,
+  payPalId: string,
+): Function {
 
-const PayPalExpressButton = (props: PropTypes) => {
-  props.setupPayPalCheckout(props.callback);
-  return <div id="component-paypal-button-checkout" className="component-paypal-button-checkout" />;
-};
+  return (dispatch, getState) => {
+    const baidCallback = baid => callback(baid, dispatch, getState);
 
+    const failureCallback = (error: string) => console.log(error);
 
-// ----- Map State/Props ----- //
+    payPalExpressCheckout
+      .setup(csrfToken, amount, billingPeriod, currency, baidCallback, failureCallback, payPalId);
+  };
+}
 
-function mapDispatchToProps(dispatch) {
 
-  return {
-    setupPayPalCheckout: (callback: Function) => {
-      dispatch(setupPayPalExpressCheckout(callback));
-    },
-  };
+// ----- Component ----- //
 
-}
+const PayPalExpressButton = (props: PropTypes) => {
+  const payPalId = 'component-paypal-button-checkout';
```

Comment:
👍 Good work on extracting the presentation details from the library integration.